pkg/encoding/v1/encoding_test.go

```diff
@@ -105,6 +115,116 @@ func TestPortType_UnmarshalYAML(t *testing.T) {
 	}
 }
 
+func TestPort_UnmarshalYAML(t *testing.T) {
+	tests := []struct {
+		Succeed bool
+		RawPort string
+		Port    Port
+	}{
+		{true, "", Port{}}, // UnmarshalYAML won't be even called for empty strings -> default value
+		{
+			true,
+			`
+port: 5000:80
+host: ""
+`,
+			Port{
+				Port: PortMapping{ContainerPort: 5000, ServicePort: 80},
+				Host: UriAddrFromString(""),
+				Path: UriPathAddrFromString(""), // path defaults to "" when host is validated/set
+			},
+		},
+		{
+			true,
+			`
+port: 5000:80
+host: "subdomain.127.0.0.1.nip.io"
+`,
+			Port{
+				Port: PortMapping{ContainerPort: 5000, ServicePort: 80},
+				Host: UriAddrFromString("subdomain.127.0.0.1.nip.io"),
+				Path: UriPathAddrFromString(""), // path defaults to "" when host is validated/set
+			},
+		},
+		{
+			false, //you have to specify host
+			`
+port: 5000:80
+path: "/admin"
+`,
+			Port{},
+		},
+		{
+			false, //you have to specify host
+			`
+port: 5000:80
+path: ""
+`,
+			Port{},
+		},
+		{
+			true,
+			`
+port: 5000:80
+host: ""
+path: ""
+`,
+			Port{
+				Port: PortMapping{ContainerPort: 5000, ServicePort: 80},
+				Host: UriAddrFromString(""),
+				Path: UriPathAddrFromString(""),
+			},
+		},
+		{
+			true,
+			`
+port: 5000:80
+host: ""
+path: "/admin"
+`,
+			Port{
+				Port: PortMapping{ContainerPort: 5000, ServicePort: 80},
+				Host: UriAddrFromString(""),
+				Path: UriPathAddrFromString("/admin"),
+			},
+		},
+		{
+			true,
+			`
+port: 5000:80
+host: "subdomain.127.0.0.1.nip.io"
+path: "/admin"
+`,
+			Port{
+				Port: PortMapping{ContainerPort: 5000, ServicePort: 80},
+				Host: UriAddrFromString("subdomain.127.0.0.1.nip.io"),
+				Path: UriPathAddrFromString("/admin"),
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run("", func(t *testing.T) {
```

Comment:
because naming is hard ;) (especially when it's suppose to make sense to others)